sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/datetimeExpressions.scala

```diff
@@ -1963,3 +1963,77 @@ case class Epoch(child: Expression, timeZoneId: Option[String] = None)
     defineCodeGen(ctx, ev, c => s"$dtu.getEpoch($c, $zid)")
   }
 }
+
+@ExpressionDescription(
+  usage = "_FUNC_(field, source) - Extracts a part of the date/timestamp.",
+  arguments = """
+    Arguments:
+      * field - selects which part of the source should be extracted. Supported string values are:
+                ["MILLENNIUM", "MILLENNIA", "MIL", "MILS",
+                 "CENTURY", "CENTURIES", "C", "CENT",
+                 "DECADE", "DECADES", "DEC", "DECS",
+                 "YEAR", "Y", "YEARS", "YR", "YRS",
+                 "ISOYEAR",
+                 "QUARTER", "QTR",
+                 "MONTH", "MON", "MONS", "MONTHS",
+                 "WEEK", "W", "WEEKS",
+                 "DAY", "D", "DAYS",
+                 "DAYOFWEEK", "DOW", "ISODOW", "DOY",
+                 "HOUR", "H", "HOURS", "HR", "HRS",
+                 "MINUTE", "M", "MIN", "MINS", "MINUTES",
+                 "SECOND", "S", "SEC", "SECONDS", "SECS",
+                 "MILLISECONDS", "MSEC", "MSECS", "MILLISECON", "MSECONDS", "MS",
+                 "MICROSECONDS", "USEC", "USECS", "USECONDS", "MICROSECON", "US",
+                 "EPOCH"]
```

Comment:
How about using parentheses for these alternatives like this?
```
      * field - selects which part of the source should be extracted. Supported string values are:
                ["MILLENNIUM" ("MILLENNIA", "MIL", "MILS"),
                 "CENTURY" ("CENTURIES", "C", "CENT"),
                 "DECADE" ("DECADES", "DEC", "DECS"),
                 "YEAR" ("Y", "YEARS", "YR", "YRS"),
                 "ISOYEAR",
                 "QUARTER" ("QTR"),
                 "MONTH" ("MON", "MONS", "MONTHS"),
                 "WEEK" ("W", "WEEKS"),
                 "DAY" ("D", "DAYS"),
                 "DAYOFWEEK" ("DOW", "ISODOW", "DOY"),
                 "HOUR" ("H", "HOURS", "HR", "HRS"),
                 "MINUTE" ("M", "MIN", "MINS", "MINUTES"),
                 "SECOND" ("S", "SEC", "SECONDS", "SECS"),
                 "MILLISECONDS" ("MSEC", "MSECS", "MILLISECON", "MSECONDS", "MS"),
                 "MICROSECONDS" ("USEC", "USECS", "USECONDS", "MICROSECON", "US"),
                 "EPOCH"]
```

Reply:
Sounds good. I will do that.